server/services/tables.js

```diff
@@ -62,6 +69,23 @@ const addTable = async (table, userCreated) => {
     return 0; //everything good
 }
 
+const leaveTable = async (name,username) => {
+    try {
+        const table = await tableSchema.findOne({ "name": name });
+        const user = await userSchema.findOne({ "username": username });
```

Comment:
again I dont understand what is name... also, nickname and username are distinct so you need also 1 of them to get 1 players (there are not 2 players with the same username or nickname) and remove them by it.

Reply:
name is the name of the table. we need it to find the correct table, username is the username of the user that leaving the table, we need to remove him from the playersOnTable . (i dont see nickname)